tests/Console/ConfigurationResolverTest.php

```diff
@@ -954,6 +954,94 @@ public function testResolveRulesWithOption(): void
         );
     }
 
+    /**
+     * Test the renamed rules.
+     *
+     * @dataProvider provideRenamedRulesCases
+     */
+    public function testResolveRenamedRulesWithUnknownRules(string $expectedMessage, string $renamedRule): void
+    {
+        $this->expectException(
+            \PhpCsFixer\ConfigurationException\InvalidConfigurationException::class
+        );
+        $this->expectExceptionMessage($expectedMessage);
+
+        $resolver = $this->createConfigurationResolver(['rules' => $renamedRule]);
+
+        $resolver->getRules();
+    }
+
+    public function provideRenamedRulesCases()
+    {
+        return [
+            [
+                'The rules contain unknown fixers: "blank_line_before_return" is a renamed rule,  did you mean "blank_line_before_statement" (note: use configuration "[\'statements\' => [\'return\']]")?'.PHP_EOL.PHP_EOL.
+                'For more info see: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.0/UPGRADE-v3.md#renamed-rules.',
+                'blank_line_before_return',
```

Comment:
where is this 3rd argument used?

Reply:
see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5887#discussion_r698745130

Reply:
> where is this 3rd argument used?

There are not a 3rd argument in the array test, it is a string concatenation, not a semicolon. I split in two lines concatenated because it is hard to read large strings in IDE or more than 80-100 chars per line. If you seem this wrongly I can change it as your desire.